### ISSUE TITLE
The `--pin` option documentation seems inconsistent

### ISSUE THREAD

**Issue by Aire-One** [luarocks/luarocks#1389]
### Please read the following before submitting:
- Please do NOT submit bug reports for questions. Ask questions in the [Gitter
  chat-room](https://gitter.im/luarocks/luarocks).
- Please do NOT submit duplicate bug reports. Look through the [issue
  tracker](https://github.com/luarocks/luarocks/issues) before submitting.

### Please fill out the following:
- **Platform:**
  - Linux (but irrelevant)

- **LuaRocks version:**
  - `luarocks --version` /usr/bin/luarocks 3.8.0

- **Configuration file:**
  - Please try to produce with the default configuration.
  - If you cannot reproduce with the default configuration, please try to find
    the minimal configuration to reproduce.
  - Upload the config to a pastebin such as gist.github.com.

Reproducible from current master after `./configure && make`

- **LuaRocks output from when the issue occurred:**
  - Use the `--verbose` flag.

- **Description:**
  - The steps you took in plain English to reproduce the problem.

Tried to use the `--pin` option.

The main luarocks help message includes a mention of this option, but it is inusable : 

```sh
$ luarocks --help
Usage: luarocks [-h] [--version] [--dev] [--server <server>]
       [--only-server <server>] [--only-sources <url>]
       [--namespace <namespace>] [--lua-dir <prefix>]
       [--lua-version <ver>] [--tree <tree>] [--local] [--global]
       [--verbose] [--timeout <seconds>] [--pin] [<command>] ...

LuaRocks 3.8.0, the Lua package manager

/usr/bin/luarocks - LuaRocks main command-line interface

Options:
[...]
   --pin                 Create a luarocks.lock file listing the exact versions
                         of each dependency found for this rock (recursively),
                         and store it in the rock's directory. Ignores any
                         existing luarocks.lock file in the rock's sources.

Commands:
[...]
```

```sh
$ luarocks --pin

[the output of `luarock` called without option or command]
```

After further investigation, this option is supposed to be used with `build` and `make` commands.
I think it should be removed from the "main help" message.

Reading #1127, the `--pin` option can also be used with the `install` command (if the installed rock is a Lua module, I'm not sure to understand what it means, looking at the code, I guess it's because when calling the `install` command on a `.rockspec` file "redirects" to the `build` command).
But this option is not part of the `luarocks install --help` message. I think it should be explicitly documented.

Thank you for your work!

**Comment by daurnimator** [comment]
https://github.com/luarocks/luarocks/blob/f33aa0125f9d8c94049c162a8fdec54582ade205/src/luarocks/cmd/build.lua#L39

I think that should  be `cmd:flag` rather than `parser:flag`.

**Comment by Aire-One** [comment]
Thank you for your input @daurnimator! I haven't noticed this while reading the source yesterday.

I'll submit a PR with a fix.
